server/src/internalClusterTest/java/org/opensearch/ratelimitting/admissioncontrol/AdmissionForClusterManagerIT.java

```diff
@@ -0,0 +1,182 @@
+import static org.opensearch.ratelimitting.admissioncontrol.settings.CpuBasedAdmissionControllerSettings.CLUSTER_ADMIN_CPU_USAGE_LIMIT;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class AdmissionForClusterManagerIT extends OpenSearchIntegTestCase {
+
+    private static final Logger LOGGER = LogManager.getLogger(AdmissionForClusterManagerIT.class);
+
+    public static final String INDEX_NAME = "test_index";
+
+    private String clusterManagerNodeId;
+    private String datanode;
+    private ResourceUsageCollectorService cMResourceCollector;
+
+    private static final Settings DISABLE_ADMISSION_CONTROL = Settings.builder()
+        .put(ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), AdmissionControlMode.DISABLED.getMode())
+        .build();
+
+    private static final Settings ENFORCE_ADMISSION_CONTROL = Settings.builder()
+        .put(ResourceTrackerSettings.GLOBAL_CPU_USAGE_AC_WINDOW_DURATION_SETTING.getKey(), TimeValue.timeValueMillis(500))
+        .put(ADMISSION_CONTROL_TRANSPORT_LAYER_MODE.getKey(), AdmissionControlMode.ENFORCED)
+        .put(CLUSTER_ADMIN_CPU_USAGE_LIMIT.getKey(), 50)
+        .build();
+
+    @Before
+    public void init() {
+        String clusterManagerNode = internalCluster().startClusterManagerOnlyNode(
+            Settings.builder().put(DISABLE_ADMISSION_CONTROL).build()
+        );
+        datanode = internalCluster().startDataOnlyNode(Settings.builder().put(DISABLE_ADMISSION_CONTROL).build());
+
+        ensureClusterSizeConsistency();
+        ensureGreen();
+
+        // Disable the automatic resource collection
+        clusterManagerNodeId = internalCluster().clusterService(clusterManagerNode).localNode().getId();
+        cMResourceCollector = internalCluster().getClusterManagerNodeInstance(ResourceUsageCollectorService.class);
+        cMResourceCollector.stop();
+
+        // Enable admission control
+        client().admin().cluster().prepareUpdateSettings().setTransientSettings(ENFORCE_ADMISSION_CONTROL).execute().actionGet();
+    }
+
+    public void testAdmissionControlEnforced() throws Exception {
+        cMResourceCollector.collectNodeResourceUsageStats(clusterManagerNodeId, System.currentTimeMillis(), 97, 99, new IoUsageStats(98));
+
+        // Write API on ClusterManager
+        assertAcked(prepareCreate("test").setMapping("field", "type=text").setAliases("{\"alias1\" : {}}"));
+
+        // Read API on ClusterManager
+        GetAliasesRequest aliasesRequest = new GetAliasesRequest();
+        aliasesRequest.aliases("alias1");
+        try {
+            dataNodeClient().admin().indices().getAliases(aliasesRequest).actionGet();
+            fail("expected failure");
+        } catch (Exception e) {
+            assertTrue(e instanceof OpenSearchRejectedExecutionException);
```

Comment:
we should assert on message/ innerCause to confirm the actual reason for rejection is AC